src/errors.c

```diff
@@ -133,8 +133,14 @@ void Error_InvalidFilterPlacement(rax *entitiesRax) {
 }
 
 void Error_SITypeMismatch(SIValue received, SIType expected) {
-	ErrorCtx_SetError("Type mismatch: expected %s but was %s", SIType_ToString(expected),
+	size_t bufferLen = 256;
+	size_t bytesWritten = 0;
+	char *buf = rm_malloc(bufferLen * sizeof(char));
+
+	SIType_ToMultipleTypeString(expected, &buf, &bufferLen, &bytesWritten);
+	ErrorCtx_SetError("Type mismatch: expected %s but was %s", buf,
 					  SIType_ToString(SI_TYPE(received)));
+	rm_free(buf);
```

Comment:
Please see if we can switch to a stack allocated buffer and remove all `realloc` checks within `SIType_ToMultipleTypeString`

Reply:
Change done [c87aa26](https://github.com/RedisGraph/RedisGraph/pull/2691/commits/c87aa2674d53357ce2cfc235e9e282f48382c362).
```bufferLen``` and ```bytesWritten``` are still used in ```SIType_ToMultipleTypeString``` to check the size of input buffer.  
Is it right?  Thanks

Reply:
bytesWritten argument was removed.